_stbt/core.py

```diff
@@ -533,14 +535,51 @@ def _load_template(template):
     if isinstance(template, numpy.ndarray):
         return _AnnotatedTemplate(template, None, None)
     else:
-        template_name = _find_path(template)
-        if not template_name or not os.path.isfile(template_name):
-            raise UITestError("No such template file: %s" % template)
-        image = cv2.imread(template_name, cv2.CV_LOAD_IMAGE_COLOR)
+        relative_filename = template
+        absolute_filename = _find_user_file(relative_filename)
+        if not absolute_filename:
+            raise IOError("No such template file: %s" % relative_filename)
```

Comment:
We should use ENOENT for "file not found". What about "invalid png file"?

* EINVAL - Invalid argument
* EIO - Input/output error
* EPROTO - Protocol error

`PIL.Image.open` raises an IOError of a single string argument, so that's no help.